### PR TITLE
Create OneBranch build and release pipeline

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Move build and release pipelines to OneBranch in Azure DevOps.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: .config/tsaoptions.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "instanceUrl": "https://msazure.visualstudio.com",
+    "projectName": "One",
+    "areaPath": "One\\MGMT\\Compute\\Powershell\\Powershell\\PowerShell Core",
+    "notificationAliases": [
+        "adityap@microsoft.com",
+        "americks@microsoft.com",
+        "annavied@microsoft.com"
+    ]
+}
```
Comment (anamnavi) [inline .config/tsaoptions.json]: add newline; nitpick 
```suggestion
}

```

File: .pipelines/PSResourceGet-Official.yml [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+            TargetFolder: $(ob_outputDirectory)
+
+        - pwsh: |
+            Write-Host "Displaying contents of signSrcPath:"
+            Get-ChildItem $(signSrcPath) -Recurse
+            Write-Host "Displaying contents of ob_outputDirectory:"
+            Get-ChildItem $(ob_outputDirectory) -Recurse
+          displayName: Get contents of dirs with signed files
+
+       # - pwsh: |
```
Comment (anamnavi) [inline .pipelines/PSResourceGet-Official.yml]: do we need this commented out part? I believe SBOM gets done by OneBranch now

File: .pipelines/tsaoptions.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "instanceUrl": "https://msazure.visualstudio.com",
+    "projectName": "One",
+    "areaPath": "One\\MGMT\\Compute\\Powershell\\Powershell\\PowerShell Core",
+    "notificationAliases": [
+        "adityap@microsoft.com",
+        "americks@microsoft.com",
+        "annavied@microsoft.com"
+    ]
+}
```
Comment (anamnavi) [inline .pipelines/tsaoptions.json]: why is there a 2nd tsaoptions.json file here?

File: global.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "sdk": {
+        "version": "6.0.418"
+    }
+}
```
Comment (anamnavi) [inline global.json]:
```suggestion
}

```

File: src/Microsoft.PowerShell.PSResourceGet.psd1 [hunk 5]
```diff
@@ -4,7 +4,7 @@
 @{
     RootModule             = './Microsoft.PowerShell.PSResourceGet.dll'
     NestedModules          = @('./Microsoft.PowerShell.PSResourceGet.psm1')
-    ModuleVersion          = '1.0.0'
+    ModuleVersion          = '1.0.0112'
```
Comment (anamnavi) [inline src/Microsoft.PowerShell.PSResourceGet.psd1]: why does this change?

**Review by anamnavi — CHANGES_REQUESTED** [review]
Just wondering why there's 2 tsaoptions.json files, rest looks great!

File: .config/tsaoptions.json (hunk 1 above)
Comment (adityapatwardhan) [reply .config/tsaoptions.json]: There is another area path that should be used:

image

File: .pipelines/PSResourceGet-Official.yml [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+      disableLegacyManifest: true
+      sbom:
+        enabled: true
+        packageName: Microsoft.PowerShell.PSResourceGet
+      codeql:
+        compiled:
+          enabled: true
+      asyncSdl: # https://aka.ms/obpipelines/asyncsdl
+        enabled: true
+        forStages: [Build]
```
Comment (adityapatwardhan) [inline .pipelines/PSResourceGet-Official.yml]:
```suggestion
        forStages: [stagebuild]
```

File: .pipelines/PSResourceGet-Official.yml [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+            files_to_sign: '**\*.psd1;**\*.psm1;**\*.ps1xml;**\Microsoft*.dll'
+            search_root: $(signSrcPath)
+
+        - task: onebranch.pipeline.signing@1
+          displayName: Sign 3rd Party files
+          inputs:
+            command: 'sign'
+            signing_profile: 135020002
+            files_to_sign: '**/Dependencies/*.dll'
+            search_root: $(signSrcPath)
```
Comment (adityapatwardhan) [inline .pipelines/PSResourceGet-Official.yml]: Before doing third party signing we should check which files are unsigned. Only those need to be third party signed. Bunch of the files under dependencies folder will be from .NET and hence will be already signed.

File: .pipelines/PSResourceGet-Official.yml (hunk 2 above)
Comment (adityapatwardhan) [reply .pipelines/PSResourceGet-Official.yml]: Remove commented code

File: .pipelines/tsaoptions.json (hunk 3 above)
Comment (adityapatwardhan) [reply .pipelines/tsaoptions.json]: This should be removed.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by adityapatwardhan** [comment]
Did you verify the built module actually works? The files are signed etc?

**Comment by alerickson** [comment]
@adityapatwardhan yup, verified files are signed and published module works

**Review by adityapatwardhan — APPROVED** [review]

**Review by anamnavi — APPROVED** [review]

**Pull request merged**